src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -909,6 +909,15 @@ internal int CalculateSyntaxOffsetInSynthesizedConstructor(int position, SyntaxT
                 return syntaxOffset;
             }
 
+            if (localKind == SynthesizedLocalKind.InstrumentationPayload)
+            {
+                // With dynamic analysis instrumentation, a type declaration can be the syntax associated
+                // with the analysis payload local of a synthesized constructor.
+                // If the synthesized constructor includes an initializer with a lambda,
+                // that lambda needs a closure that captures the analysis payload of the constructor.
+                return 0;
```

Comment:
Zero is not a magic number here--it's the correct offset from the start of the provided syntax node. The next iteration will make this look less mysterious.